### DISCUSSION TITLE
Fields being overwritten with Polylang Plugin

### DISCUSSION THREAD

**Discussion by Shelob9** [pods-framework/pods#6589] [Ideas]
A user in [this support forum thread](http://pods.io/forums/topic/problem-with-polylang/) reports the following issue using using Polylang with Pods:

"I have Polylang working with 3 different language without problems with post and pages translations.
I create a new pod (page) with the title, the text and 4 fields Wysiwyg and activate Polylang."

WordPress 3.7.1, Polylang 1.1.6 and Pods 2.3.18.

**Comment by unknownnf** [comment]
Fields can be overwritten with pods in Polylang if Custom Fields synchronization is enabled. This happens to both meta based and table based fields afaik. Also creating a new translation with existing values in the fields with that setting on would create a new page, with empty fields and empty all fields for all translations.

**Comment by sc0ttkclark** [comment]
@unknownnf what do we do here?

**Comment by unknownnf** [comment]
Well the problem is that polylang/wpml get the raw meta value and sync that, however our form gets the value of a new pods object which is empty.

The more I try to fix this the more I think this should be a separate plugin to handle multilingual stuff. Then we can just hook into value or input functions and get whatever we need depending on the settings.

Would be a lot more elegant than hacking the base plugin to support this.

**Comment by sc0ttkclark** [comment]
Sounds like a good idea

**Comment by unknownnf** [comment]
Yeah but this would require more re-factoring for fields to have a raw value function call or something along those lines. Currently we always pass the value to a field with function calls, which probably means a lot of hooks and whistles to get this going, this is probably going to cause a lot of problems further down the line.

**Comment by sc0ttkclark** [comment]
Pods::raw exists, and Pods::field takes a 'raw' parameter (Pods::raw just maps to that), does that help?

**Comment by fuldauer** [comment]
Hi there! Have you advanced any in this issue? I'm stuck in a project because of this and I'd like to know, because I had already lost all hope, but because of all the work invested already I'm giving it a another try.

**Comment by JoryHogeveen** [comment]
From what I read here it's about syncing metadata between translations (for WPML and Polylang).

Though I have started with a sync plugin for media (attachments), this functionality isn't very easy to implement for all types since you're also dealing with possible non-existend translations etc.

https://github.com/JoryHogeveen/pods-polylang-sync-meta

This plugin currently only syncs **attachments** and automatically adds translations if they don't exist.

If anyone want's to help in creating this functionality for all relationship types, please let me know!

**Comment by quasel** [comment]
@JoryHogeveen maybe make a master list of i18n issues? to see what's open or triage that stuff? so may unresolved issues around that… and all the plugins have advanced in the last year++ ;)

**Comment by JoryHogeveen** [comment]
> and all the plugins have advanced in the last year

This still isn't related to syncing relational data. i18n plugins will never support this since it's not a feature from WP. That is why I started with an addon for Pods. If you use ACF for example you will need a different tool/plugin to sync any relational data.
